### PR TITLE
Bugfix: Attach correct link to "levitate detect breaking changes"-message included in PR

### PR DESCRIPTION
**Pull request by mckn**
**What is this feature?**
The link that gets added to the "Detect breaking changes" message posted by Levitate pointed towards the wrong build step. This PR will change so it should point to the correct job and step.

**Why do we need this feature?**
To make it easier for developers of Grafana to view the detect breaking changed output.

**Who is this feature for?**
Developers of grafana/grafana

**Which issue(s) does this PR fix?**:
No issue reported

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/workflows/scripts/pr-get-job-link.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
     const { owner, repo } = context.repo;
     const url = `https://api.github.com/repos/${owner}/${repo}/actions/runs/${context.runId}/jobs`
-    const result = await github.request(url)
-    const link = `https://github.com/grafana/grafana/runs/${result.data.jobs[0].id}?check_suite_focus=true`;
-
+    
+    const result = await github.request(url);
+    const job = result.jobs.find(j => j.name === name);
+    const stepIndex = job.steps.findIndex(s => s.name === name);
+    const link = `${job.html_url}#step:${stepIndex + 1}:1`;
```
Comment (tolzhabayev) [inline .github/workflows/scripts/pr-get-job-link.js:10]: I think this file then has to change too?

- https://github.com/grafana/grafana/blob/main/.github/workflows/detect-breaking-changes-build.yml#L150
- https://github.com/grafana/grafana/blob/main/.github/workflows/detect-breaking-changes-build.yml#L110-L111

File: .github/workflows/scripts/pr-get-job-link.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
 
 module.exports = async ({ github, context, core }) => {
+    const name = "Detect breaking changes";
```
Comment (tolzhabayev) [inline .github/workflows/scripts/pr-get-job-link.js:3]: Might be better to have the name as input but that's just a nit.

File: .github/workflows/scripts/pr-get-job-link.js (hunk 1 above)
Comment (mckn) [reply .github/workflows/scripts/pr-get-job-link.js:10]: Ah, thanks! Totally missed that the step got added there.

File: .github/workflows/scripts/pr-get-job-link.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
     const { owner, repo } = context.repo;
     const url = `https://api.github.com/repos/${owner}/${repo}/actions/runs/${context.runId}/jobs`
-    const result = await github.request(url)
-    const link = `https://github.com/grafana/grafana/runs/${result.data.jobs[0].id}?check_suite_focus=true`;
-
-    core.setOutput('link', link);
+    const result = await github.request(url);
+    const job = result.jobs.find(j => j.name === name);
+    
+    core.setOutput('link', job.html_url);
```
Comment (tolzhabayev) [inline .github/workflows/scripts/pr-get-job-link.js:8]:
```suggestion
    core.setOutput('link', `${job.html_url}?check_suite_focus=true`);
```

maybe to keep it the same way it was before? I think it changes the sidebar a bit / makes it more compact.

**Review by leventebalogh — APPROVED** [review]
Nice 👏

File: .github/workflows/scripts/pr-get-job-link.js (hunk 3 above)
Comment (mckn) [reply .github/workflows/scripts/pr-get-job-link.js:8]: For me that one doesn't change the UI but I'm happy to add it if needed.

Comment (tolzhabayev) [reply .github/workflows/scripts/pr-get-job-link.js]: Here is an example with and without (look at the sidebar on the left):
- https://github.com/grafana/grafana/runs/10088512148?check_suite_focus=true#step:7:1 (with)
- https://github.com/grafana/grafana/runs/10088512148?#step:7:1 (without)

**Pull request merged**